### PR TITLE
Add elasticsearch formatter to support bulk api

### PR DESCRIPTION
**Pull request by ehazlett**
This adds an Elasticsearch formatter to support the bulk API.

### CODE REVIEW

**Comment by ehazlett** [comment]
@i3149 I included the "instrumentation" and "provider" fields from the `json` formatter.  I'm not sure if we need them but I included them anyway.

**Comment by i3149** [comment]
> @i3149 I included the "instrumentation" and "provider" fields from the `json` formatter. I'm not sure if we need them but I included them anyway.

Might as well keep these. Not needed but can provide some context to the data.

File: pkg/formats/elasticsearch/elasticsearch.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		ContextL: logger.NewContextLFromUnderlying(logger.SContext{S: "elasticsearchFormat"}, log),
+	}
+
+	return jf, nil
+}
+
+func (f *ElasticsearchFormat) To(msgs []*kt.JCHF, serBuf []byte) (*kt.Output, error) {
+	msgsNew := make([]map[string]interface{}, 0, len(msgs))
+	for _, msg := range msgs {
+		if msg.EventType == kt.KENTIK_EVENT_SNMP {
```
Comment (i3149) [inline pkg/formats/elasticsearch/elasticsearch.go:39]: Probobly can take this if out, just pass all types along.

File: pkg/formats/elasticsearch/elasticsearch.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		msgsNew = append(msgsNew, mm)
+	}
+
+	esBulkData := ""
+	for _, m := range msgsNew {
+		data, err := serialize(m)
+		if err != nil {
+			return nil, err
+		}
+		esBulkData += data
```
Comment (i3149) [inline pkg/formats/elasticsearch/elasticsearch.go:53]: I'm not entirely sure about how go handles string append, but generally faster to make esBulkData a slice of strings and then join with `""` at the end?

File: pkg/formats/elasticsearch/elasticsearch.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+func (f *ElasticsearchFormat) Rollup(rolls []rollup.Rollup) (*kt.Output, error) {
+	// serialize rolls
+	esBulkData := ""
+	for _, m := range rolls {
+		data, err := serialize(m)
+		if err != nil {
+			return nil, err
+		}
+		esBulkData += data
```
Comment (i3149) [inline pkg/formats/elasticsearch/elasticsearch.go:93]: Same, avoid the looping string append.

File: pkg/formats/format.go [hunk 4]
```diff
@@ -25,21 +26,24 @@ type Formatter interface {
...
+	FORMAT_NRM                  = "new_relic_metric"
+	FORMAT_SPLUNK               = "splunk"
 )
 
 func NewFormat(format Format, log logger.Underlying, compression kt.Compression) (Formatter, error) {
 	switch format {
 	case FORMAT_AVRO:
 		return avro.NewFormat(log, compression)
+	case FORMAT_ELASTICSEARCH:
+		return elasticsearch.NewFormat(log)
```
Comment (i3149) [inline pkg/formats/format.go:46]: Do we want / need to handle compression here also? Not sure you ever want to gzip sending into ES but its an option.

**Review by i3149 — COMMENTED** [review]
Looks good. Would we also want to extend the http sink to allow methods other than POST? Not sure if needed or not.

**Comment by ehazlett** [comment]
> Would we also want to extend the http sink to allow methods other than POST? Not sure if needed or not.

We don't need it for the ES bulk API so I think we could have a follow up with that.

**Review by i3149 — DISMISSED** [review]
LGTM

**Review by i3149 — APPROVED** [review]
Continues to look good, with compression added :rocket:

**Pull request merged**